sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ClientSideRequestStatistics.java

```diff
@@ -99,6 +99,11 @@ public void recordResponse(RxDocumentServiceRequest request, StoreResultDiagnost
         storeResponseStatistics.storeResult = storeResultDiagnostics;
         storeResponseStatistics.requestOperationType = request.getOperationType();
         storeResponseStatistics.requestResourceType = request.getResourceType();
+        if (request.requestContext.sessionToken != null) {
+            storeResponseStatistics.requestSessionToken = request.getHeaders().get(HttpConstants.HttpHeaders.SESSION_TOKEN);
+        } else {
+            storeResponseStatistics.requestSessionToken = request.getOriginalSessionToken();
```

Comment:
request.getHeaders().get(HttpConstants.HttpHeaders.SESSION_TOKEN) itself should be enough